src/WiFi.cpp

```diff
@@ -723,4 +733,54 @@ void WiFiClass::refresh(void)
 	m2m_wifi_handle_events(NULL);
 }
 
+uint8_t WiFiClass::ping(const char* hostname, uint8_t ttl){
+	IPAddress ip;
+	if (hostByName(hostname, ip) > 0)
+		return ping(ip, ttl);
+	else
+		return WL_PING_ERROR;
+};
+
+uint8_t WiFiClass::ping(const String &hostname, uint8_t ttl){
+	return ping(hostname.c_str(), ttl);
+};
+
```

Comment:
@sandeepmistry if we drop this overload, any String will need extra .c_str() call:
ping(hostname.c_str());

Also passing string constants will not be possible:
ping("www.google.com"); // This will not compile, but is a handy call

I think this is unhandy for beginners.  For example hostByName has "char*" and "String" overloads.
I suggest keeping all the three overloads. They do not add much overhead. Anyone can choose what to use. What do you think?